src/components/LocationListener.tsx

```diff
@@ -0,0 +1,102 @@
+import {Component} from 'react'
+import BackgroundGeolocation, {
+  Location,
+} from 'react-native-background-geolocation'
+import LocationStorageContainer from '../containers/LocationStorageContainer'
+
+interface Props {
+  locationStorage: LocationStorageContainer
+}
+
+class LocationListener extends Component<Props> {
+  // TODO: prompt the user for permission, replace default settings
+  componentDidMount() {
+    // This handler fires whenever bgGeo receives a location update.
+    BackgroundGeolocation.onLocation(this.onLocation, this.onError)
+    // This handler fires when movement states changes (stationary->moving; moving->stationary)
+    BackgroundGeolocation.onMotionChange(this.onMotionChange)
+
+    // This event fires when a change in motion activity is detected
+    BackgroundGeolocation.onActivityChange(this.onActivityChange)
+
+    // This event fires when the user toggles location-services authorization
+    BackgroundGeolocation.onProviderChange(this.onProviderChange)
+
+    BackgroundGeolocation.ready(
```

Comment:
Looks like there is still a lot of placeholder text in here. I know this is blocking and other work builds off this branch in regards to location things so I'm okay if were planning on coming back over this next and cleaning it up.